metaflow/runner/metaflow_runner.py

```diff
@@ -326,6 +326,7 @@ def run(self, **kwargs) -> ExecutingRun:
                 show_output=self.show_output,
             )
             command_obj = self.spm.get(pid)
+            command_obj.sync_wait()
 
             return self.__get_executing_run(attribute_file_fd, command_obj)
```

Comment:
I guess `command_obj.sync_wait()` for Runner should happen inside `self.__get_executing_run` since we wanna do it after `content = handle_timeout(...)`